### PR TITLE
yoda: 1.6.5 -> 1.6.6, provide version with ROOT

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/physics/yoda/default.nix [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   src = fetchurl {
     url = "http://www.hepforge.org/archive/yoda/YODA-${version}.tar.bz2";
-    sha256 = "1i8lmj63cd3qnxl9k2cb1abap2pirhx7ffinm834wbpy9iszwxql";
+    sha256 = "088xx4q6b03bnj6xg5189m8wsznhal8aj3jk40sbj24idm4jl5yg";
   };
 
   pythonPath = []; # python wrapper support
 
-  buildInputs = with python2Packages; [ python numpy matplotlib makeWrapper ];
+  buildInputs = with python2Packages; [ python numpy matplotlib root makeWrapper ];
```
Comment (Mic92) [inline pkgs/development/libraries/physics/yoda/default.nix]: What is the disadvantage of having this enabled by default?

Comment (veprbl) [inline pkgs/development/libraries/physics/yoda/default.nix]: ROOT adds 433 mb to closure and the only practical reason to enable it is to have root2yoda and yoda2root utilities. The reverse dependencies of YODA don't care about those. So yoda-with-root is for using interactively when user need those extra conversion utilities (most often you don't need those, actually).

**Comment by rasendubi** [comment]
```
trying https://yoda.hepforge.org/hg/yoda/rev/3dbc8927e715?style=raw
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1017    0  1017    0     0   2128      0 --:--:-- --:--:-- --:--:--  3852
output path ‘/nix/store/cy0yrq6gr61j108cpzsmbjrinrq2fhzc-3dbc8927e715?style=raw’ has sha256 hash ‘02rm34z9lbab66p7gpij1
2qwdph5fddpksg80qz0m537wjwy2ddy’ when ‘0rjgyfxwl5pps2n3fhjnyiwjn20fmwxvpbvavp0jnh8j8pjclrnj’ was expected
cannot build derivation ‘/nix/store/95j49nnzpf147dpbyr3cfcw64m6kd69w-yoda-1.6.6.drv’: 1 dependencies couldn't be built
```

**Comment by veprbl** [comment]
I guess this is one of those instances when nix-prefetch-url returns something completely different from what nix-build sees. I fixed the hashes. @rasendubi Sorry and thank you!

**Comment by rasendubi** [comment]
`fetchpatch` does additional processing of the downloaded patches, so the hash is usually different from `fetchurl`.

**Comment by veprbl** [comment]
I did not realize that. Thanks. Also in my tests ```nix-prefetch-url -A``` doesn't work with ```fetchpatch``` either.

**Comment by vcunat** [comment]
I believe `nix-prefetch-url -A` would work fine with `fetchpatch`, but it won't work with lists &ndash; and you typically use it in the `patches` list.

**Comment by veprbl** [comment]
@vcunat In my tests I was creating a separate attribute to overcome the list problem. I believe that still didn't get a correct hash even after that.

**Pull request merged**